text/0002-bundler-2.md

```diff
@@ -0,0 +1,43 @@
+# Bundler 2 RFC
+
+This document outlines the changes proposed for the Bundler v2 release. The main goal of this release will be to remove
+support for Ruby < 2.0, remove and deprecate existing functionality and introduce any changes to existing behaviour.
+
+## Removed Functionality
+* The `bundle viz` and `bundle show` commands. The `viz` command will be moved to a Gem/Bundler Plugin and `show` will be removed in favour of `bundle info` and `bundle list`
+* The `bundle_ruby` executable [#3489]
+* Capistrano integration in `bundler/capistrano.rb` [#3551]
+* remove `--no-cache`, `--binstubs`, `--path`, `--with`, `--without`, `--system` options from `bundle install` [#3555, #3774, #3827, #3937, #3939, #3994]
+* The `bundle cache` command implementation is removed [#4008]
+* No longer checking for older version of Bundler in the $LOAD_PATH in `exe/bundler` [#3662]
+* The `disable_shared_gems` option [#3999]
+* The `bundle console` command [#4034]
+* RubyGems Aggregate & support transitive source pinning [#4714]
+
+## New Functionality
+* Add `--cache` option to `bundle install` [#3555]
+* Summarize the number of using gems [#3872]
+* cache downloaded gems across applications [#3983, #5782]
+* Add `--shebang` option to `bundle binstubs` [#4070]
+* Specify ruby version when using the gemspec method [#4321]
+
+## Breaking Changes
+* remove support for versions of Ruby < 2.0 [#3842]
+* The `install` command no longer remembers the given options, ie: `bundle install --path vendor` [#3955]
+* Updating all gems in the bundle through the `bundle update` command will now require the `--all` option. [#2646]
+* `bundle update` now requires a gem name unless the `--all` option is provided.
+* `bundle install` will install gems into `.bundle/gems` by default [#3784]
+* Don't create gem home before setting it [#2884]
+* The `--path` option for `bundle install` will raise an error notifying the user to use `bundle config path <path/to/bundle>` instead.
+* `bundle` without any subcommand will print the help page instead of installing gems [#3831]
+* The `Gemfile` and `Gemfile.lock` will move to `gems.rb` and `gems.locked` [#3834]
+* Allow mis-matches in bundler requirements [#3871]
+* The `bundle package` command is now `bundle cache` [#4008]
+* Use `bundle gem --exe` as primary option name, instead of `--bin` [#4261]
```

Comment:
This change seems to be applied to the master branch since Bundler 1.12.0.

https://github.com/bundler/bundler/commit/95d65d5870da321d78505612a325d81f19b23d9a

Is it to remove `--bin` option from`bundle gem` command in Bundler 2.0?

https://github.com/bundler/bundler/pull/4226#issuecomment-174200451